### DISCUSSION TITLE
v2.0.7

### DISCUSSION THREAD

**Discussion by the1812** [the1812/Bilibili-Evolved#2640] [分享 / Share]
**欢迎来到 Bilibili Evolved v2.0**

经过 6 个月的公测, Bilibili Evolved v2.0 已经基本准备完毕, 是时候正式推出了. 正好最近 b 站播放器页面又改版, 也懒得再折腾 v1 了, v1 现在已经变得越来越臃肿, 添加新功能 / 修改老功能都很麻烦, 还有很多奇奇怪怪的遗留 bug. 所以本次发布起, 将完全停止对 v1 的支持.

新版本中有以下特点:
- 引入现代化前端工具 (webpack, Babel, PostCSS 等)
- 源代码 TypeScript 全覆盖 (webpack 等配置文件除外)
- 自带 UI 组件库 (基于 Vue)
- 移除对 jQuery 和 Vuex 的依赖
- 全新的架构设计, 脚本体积大幅缩小
- 不再有离线版, 你可以自行控制功能的更新

对于预览版用户, 本次更新的内容为:

- 新增 `下载视频 - MPV 输出支持` 插件. (PR #2605 by [diannaojiang](https://github.com/diannaojiang))
- 修复 `v1 风格设置面板` 中功能面板弹窗的方向错误. (#2604)

**安装方式:**

脚本不会由脚本管理器自动更新到 v2, 更新提示里也不会直接让你安装更新. 因为 v2 的安装和使用方式和 v1 有很大区别, 即便你是 v1 的老用户, 也强烈建议你重新阅读 [README 的"设置"小节](https://github.com/the1812/Bilibili-Evolved/blob/master/README.md#设置).

这里也再重新强调一下安装前的注意事项:
- **Tampermonkey beta v4.14.6147 / 6148 版有 bug, 不能使用 v2.** (https://github.com/Tampermonkey/tampermonkey/issues/1381)
- **请注意浏览器的版本, v2 提高了对浏览器的要求: Chrome 84+ / Firefox 80+ / Edge 84+ / Safari 14.1+ ; 升级前务必检查浏览器版本是否支持.**
- **有一小部分功能在 v2 中还未实现 (下面的"功能差异"里有写), 如果非常需要这些缺少的功能, 请不要升级, 对于 v1 中弹出的升级提示, 选择 `不再提示` 即可.**
- 做好觉悟, 脚本开启后不能使用弹幕点赞和举报, 全景视频不能用鼠标拖拽视角 (只能用键盘操作), 对性能也有较大影响.
- 新版本一旦正式发布, 就不再对旧版本做任何技术支持.
- 使用外部网站的链接时 (如将下载任务发送到自己的服务器 / 使用第三方链接安装组件等) 可能会提示"脚本试图访问跨域资源", 请选择"始终允许".
- 可能无法很好地适应窄屏幕, 请尽量以 1400 x 800 以上的逻辑分辨率使用此脚本.

首先备份 v1 设置 (之后会用到), 然后彻底卸载 v1 (脚本管理器有内置回收站的也要在回收站里彻底删除), 避免遗留数据继承至 v2 造成奇奇怪怪的问题.

根据自身网络情况, 选择一个安装源来安装: (不要在 Release 里下 zip 包)
- jsDelivr: https://cdn.jsdelivr.net/gh/the1812/Bilibili-Evolved@master/dist/bilibili-evolved.user.js
- GitHub Raw: https://raw.githubusercontent.com/the1812/Bilibili-Evolved/master/dist/bilibili-evolved.user.js

全新安装的 v2 不包含任何实质性的功能, 它本身更接近于一个功能管理器, 你可以装个 `v1 设置迁移` 导入 v1 的设置 (顺便还能熟悉下怎么装功能), v1 开着的功能会在 v2 中自动安装, 详情见[此文档](https://github.com/the1812/Bilibili-Evolved/blob/preview/doc/v1-migrate.md).

迁移完成后, 除了下面说的功能差异, 其他功能都应该自动装好并继承了 v1 的设置, 可以正常使用了. 此时你可以将 `v1 设置迁移` 直接卸载.

> 自定义顶栏因为重构过, 迁移完后有些项目的顺序可能会变, 直接去设置里调一下顺序就行了.

**功能差异:**

- 暂不支持 v1 的 `简化首页` 功能, 在 v2 中 `简化首页` 等于 v1 的 `首页过滤` 功能. v1 的 `简化首页` 由于代码设计过于糟糕, 很难再进一步扩充功能, 我计划在 v2 发布后尽快重写这个功能, 届时会比 v1 更加美观和强大.
- 下载视频暂不支持 `ffmpeg 命令生成` 和 `课程下载`.
- 暂不支持 v1 的 `界面翻译` 功能, 以后可能会做英语支持.
- 下载视频删除了 `直接下载` 的功能 (这个是 **直接删除** 不是 **暂不支持**, 因为遗留 bug 过多), 改为 Toast 输出, 区别是可以直接调用浏览器下载, 没有内存大小限制, 但不能给文件自动命名. (aria2 输出依然可以自动命名)

暂不支持的那几个, 后续实现时的优先级从上至下排列. 虽然这几个功能还没跟上, 但是也有其他一些功能在 v2 中得到了改进, 例如 `快捷键扩展` 支持了在图形化界面中自定义配置.

**策略调整:**

- 有更高的更新频率, 相对每次较少的更新量.
- 以下功能由于我精力有限, 转为低优先级支持, 如果对这些功能的开发感兴趣, 欢迎[参与开发](https://github.com/the1812/Bilibili-Evolved/blob/preview/CONTRIBUTING.md):
  - 夜间模式 (darkMode)
  - 默认播放器模式 (defaultPlayerMode)
  - 播放器置顶 (playerOnTop)
  - 播放器定位 (playerFocus)

参与开发的文档说明请参考[代码贡献指南](https://github.com/the1812/Bilibili-Evolved/blob/preview/CONTRIBUTING.md), v2 虽然提高了使用门槛, 但是降低了开发门槛, 不再需要像 v1 一样装 .NET Core 才能编译了. 只需要 VSCode, Node.js 和 yarn, 熟悉前端技术的话上手这个项目应该不成问题.

如有其他疑问, 请在这个 Release 对应的讨论区里回复.

<hr /><em>This discussion was created from the release <a href='https://github.com/the1812/Bilibili-Evolved/releases/tag/v2.0.7'>v2.0.7</a>.</em>

**Comment by VergilGao** [comment]
恭喜恭喜，可惜没有备份V1数据就更新到V2了，刚才调了半天，但是最后效果还是蛮不错的。

**Reply by ChaXxl** [reply]
你再把v1的安装上，设置还能导出的，是以前的设置

**Reply by Yaklo** [reply]
怎么装旧版的啊，离线版没设置啊

**Reply by Geeyun-JY3** [reply]
> 怎么装旧版的啊，离线版没设置啊

往下翻一下：https://github.com/the1812/Bilibili-Evolved/discussions/2640#discussioncomment-1789982

**Comment by kethoven** [comment]
通过对比v1和v2版本。感觉v2还有一些小问题
1.v2版本没有提示了吗？
就是类似于稍后再看库存已满。
2.自定义顶栏，动态点击“稍后再看”，再去取消，图标不能够实时变化。 顶栏的“稍后再看”和“收藏”，不能够实时更新。
3.播放某个up主的全部视频页面没有了“快速收藏”图标
如https://www.bilibili.com/medialist/play/20553583?business=space&sort_field=pubtime&tid=0
v1版本是有快速收藏功能的。

**Reply by the1812** [reply]
1 是有的吧, 比如互动视频添加稍后再看是会报错的

其他的后续可以再修复下

**Reply by kethoven** [reply]
> 1 是有的吧, 比如互动视频添加稍后再看是会报错的
> 
> 其他的后续可以再修复下

我这边顶栏动态里面，添加稍后再看，库存满了，没提示。

**Comment by XCSenior** [comment]
加油

**Reply by XCSenior** [reply]
按照大佬的文档，导入了以前v1版本的设置，一下子爷青回了(￣▽￣)／

**Comment by Brandon0651** [comment]
这个下载连接真的打得开吗，试了好多节点。。

**Comment by WanCixiong** [comment]
如何重新下载V1版本，V2目前实在是不习惯，好多功能我都迷失了

**Reply by the1812** [reply]
https://cdn.jsdelivr.net/gh/the1812/Bilibili-Evolved@master/bilibili-evolved.offline.user.js

**Reply by WanCixiong** [reply]
谢谢

**Reply by RingingVolcano** [reply]
建议还是尽快过渡适应
因为B站官方的改版调整 V1停止支持后可能很快就会有各种各样的问题

**Comment by LiangCha01** [comment]
好用！

**Comment by yuan21c** [comment]
![1561354312-AUELlbhWna](https://user-images.githubusercontent.com/22002305/145693379-e9c9f875-2d01-4e2e-adb6-dba2e641724d.gif)

**Comment by leelh2433** [comment]
等v2简化首页功能出了我在升吧，实在是不习惯b站原先的页面。

**Comment by Witzzz** [comment]
一人血书希望简化首页尽快上线！！（另外b站自带的按住→键三倍速播放希望可以像v1一样保留:P） 最后磕头感谢大佬的无私奉献，非常非常感谢！！！！！！！！！！

**Reply by Geeyun-JY3** [reply]
血书也是
> 问就是 Coming Soon.

参考：#2138、#2352
长按 <kbd>→</kbd> 倍速好像还没适配，没了应该是你开了 `快捷键扩展`

**Reply by JianPpan** [reply]
适配了吧，我这边使用正常的

**Reply by Witzzz** [reply]
喔！！！真的诶，关掉快捷键扩展就可以手动三倍速了好耶

**Comment by wenjiazhou123** [comment]
大佬,我有个小问题,v2的扩展倍速功能找不到对应的组件,通过v1设置导入可以生效扩展倍速,但是如果不通过配置导入,就无法安装倍速扩展组件.同时,通过v1导入的扩展倍速功能,在v2设置面板也找不到对应的开关

**Reply by Geeyun-JY3** [reply]
`扩展倍速菜单` 的开关目前在 `倍速记忆` 里。有考虑拆分的：#2623

**Comment by x1renn** [comment]
原先版本可以让鼠标在首页时预览动态内容实时更新，现在需要刷新页面才能更新了，v2里现在有这个功能吗，还是后续会做？

**Reply by Geeyun-JY3** [reply]
这说的是 v1 的 `简化首页` 功能还是自定义顶栏

**Reply by x1renn** [reply]
忘记了，一直觉得这个功能很实用，但是现在好像没有看到相应的组件

**Reply by Geeyun-JY3** [reply]
是作用于首页还是顶栏的功能都分不出吗……

**Comment by Edelweissor** [comment]
期待简化页面，大佬nb

**Comment by SpectreAX** [comment]
番剧、电影这些版权内容是不能下载了吗

**Reply by the1812** [reply]
还是那句话, 你能看就能下, 不能看就是不能

**Reply by SpectreAX** [reply]
都能正常观看的，就是不能下载，回头重装脚本试试，谢谢大佬

**Reply by SpectreAX** [reply]
弹出下载对话框后需等待几秒钟并且手动切换下载格式（flv切dash再切回flv）后可下载，问题已解决，谢谢

**Comment by a315152269** [comment]
多谢作者，很好用，用了好久了。

**Comment by LONG-STUDIO** [comment]


**Reply by the1812** [reply]
你在教程下面求教程?

**Reply by Geeyun-JY3** [reply]
你甚至没搞清楚什么叫油猴（

**Reply by LONG-STUDIO** [reply]
别骂了别骂了，我研究一下研究一下，我应该可以的

**Comment by JianPpan** [comment]
这是来自QQ邮箱的自动回复邮件。您好，来信已经收到，将尽快给您回复。

**Comment by Revicess** [comment]
为什么更新后首页没有v1版的美化了？只看到一个简化首页样式

**Reply by Geeyun-JY3** [reply]
认真再看一次更新日志

**Reply by ChaXxl** [reply]
看完所有日志和discussion再发言啦，作者大大已经说过了简化首页后面会更新出来的 我之前也提过了#2620

**Comment by cjf0523** [comment]
建议把倍速扩展和倍速记忆分开，我只想用倍速扩展的功能，并不需要倍速记忆，V1的时候就挺好

**Reply by Geeyun-JY3** [reply]
重复：#2623

**Comment by Aidhyh** [comment]
为什么我的脚本启用后, 在网页左侧中央没有有功能面板和设置面板的入口，这该如何是好？

**Reply by Geeyun-JY3** [reply]
你这什么信息也没有很难帮你判断。

检查一下浏览器版本是否符合要求，善用 GitHub 搜索找找有没有类似情况的 issue 或 discussion，再不行就开 issue 按要求填信息。

**Comment by lyjhyjyes** [comment]
您好，win10 chrome96 tamperkonkey 4.13 bilibili evolved v2.0.8-10-g16157b4de，已搜索水印关键词，只有关于直播水印的去除。请问能否添加视频右上角的bilibili水印的去除功能。
https://greasyfork.org/zh-CN/scripts/413923 之前多个类似的去除水印脚本失效了，官方代码升级

**Reply by Geeyun-JY3** [reply]
你确定这不只是个解析 VIP 的脚本？解析后的视频源无水印而已。

**Comment by lyjhyjyes** [comment]
不是啊。我没用解析功能，直接看原视频，没有水印。b站腾讯爱奇艺都是无水印

Geeyun-JY3 ***@***.***> 于2021年12月18日周六 11:30写道：

**Reply by Geeyun-JY3** [reply]
什么视频水印不是内嵌的啊？普通 UP 投稿的视频有没有 B 站样式的水印取决于 UP 的设置，番剧电影的不都内嵌的吗？给个例子？

另外请来 GitHub 回复，不要通过邮件回复，保证讨论串连续。

**Reply by lyjhyjyes** [reply]
> 什么视频水印不是内嵌的啊？普通 UP 投稿的视频有没有 B 站样式的水印取决于 UP 的设置，番剧电影的不都内嵌的吗？给个例子？
> 
> 另外请来 GitHub 回复，不要通过邮件回复，保证讨论串连续。

sorry，不知道邮箱回复会开新楼层。
https://www.bilibili.com/video/BV1f44y1E7mZ 比如这个视频有up主水印+b站水印。我无法区分b站水印是否内嵌。
脚本不是很懂

**Reply by Geeyun-JY3** [reply]
你确定这种取决于 UP 主设置的水印之前有脚本能去掉？
![image](https://user-images.githubusercontent.com/49396421/146630729-583be800-febf-432a-a606-104a91609d4f.png)

**Reply by MC-dusk** [reply]
如果水印是上传后压制进去的，是去不掉的，但有特殊方法可以拿到无水印视频，详见https://github.com/nilaoda/BBDown

**Comment by MC-dusk** [comment]
有一个小问题，v1备份设置只有7k，v2备份的设置高达1.3M，请问是正常现象吗

**Reply by the1812** [reply]
正常

**Comment by cjf0523** [comment]
使用倍速播放后再按键盘右方向键加速，视频倍速播放就会恢复成1倍速
